base/number.jl

```diff
@@ -67,9 +67,12 @@ function getindex(x::Number, I::Integer...)
     @boundscheck all([i == 1 for i in I]) || throw(BoundsError())
     x
 end
+# some code treats numbers as collection-like
 first(x::Number) = x
 last(x::Number) = x
-copy(x::Number) = x  # some code treats numbers as collection-like
+copy(x::Number) = x
+collect(x::Number) = x
```

Comment:
This may be worth calling out and thinking through a bit more.  I expected `collect(1)` to return `[1]` since numbers are iterable "collections". In fact, right now `collect(1) = fill(1)` (that is, a zero-d array).

 I get why this change is needed, but I'm not entirely sold that it's the right answer.

Reply:
Agreed, I'm not entirely sold on it either :).

Reply:
I'm also -1 on this definition.